### PR TITLE
modified main.rb file in game_logic branch because of raised conflicts

### PR DESCRIPTION
**Pull request by MahmoudBakr23**
- Added player class
- Added board class
- Updated the game logic in the main.rb file

Note:
We copied the code from an old game_logic branch that we created and pasted it in another new one to avoid raised conflicts between the branches

### CODE REVIEW

**Comment by davidmrtz-dev** [comment]
## Hi @MahmoudBakr23  . 👨‍🚀
### Very good job so far.

However, there are some issues to be implemented/fixed, please check the comments pointed below.  👨‍💻  ⬇️ 

****
**Please make the changes required and submit for re-review once you are ready!** 
****
### Status: Changes required. ♻️  
****


**[TSE:  David Elí]** 
Slack: @ David Eli Martinez Garcia

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      (0..@board.length - 1).each do |j|
+        @board[i][j] = player_mark if @board[i][j] == move
+      end
+    end
+  end
+
+  def display_board
+    (0..@board.length - 1).each do |i|
+      (0..@board.length - 1).each do |j|
+        print @board[i][j].to_s + ' '
```
Comment (davidmrtz-dev) [inline lib/board.rb]: - [ ] No puts/print and get.chomp outside main.rb file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png)) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

File: bin/main.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  move = valid_number(move)
+  move = check_invalid_slot(move, board)
+  board.booked_slots.push(move)
+  board.mark_on_board(move, player_turn.mark)
+  if move_counter >= 5 && board.win_check(player_turn.mark)
+    puts "#{player_turn.name} is winner"
+    return false
+  end
+
+  if move_counter >= 5 && move_counter == 9
```
Comment (davidmrtz-dev) [inline bin/main.rb]: There's an issue with a draw, After a draw game, the game still running and requesting a move from the next player.
You can ask the user if they want to initialize a new game.
Screen Shot 2020-07-20 at 6 32 42 PM

File: lib/player.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Player
+  attr_accessor :name, :mark
```
Comment (davidmrtz-dev) [inline lib/player.rb]: - [ ] The student uses attr_readers/attr_writers. Only uses attr_accessors if they really need it and must be aware that by using it they allow writing that property from outside the class. That variable can be set to anything.([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png)) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

I notice you are accessing to the name and mark just to read the value, please, instead use a reader, it's better to use an accessor when you have to change and read the values of the variables.

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

Good use of private methods. 👌

**STATUS: APPROVED ✅**

Don't forget to fill the project completion form👍
Good luck with the next project.

HAPPY CODING👨‍💻

**Pull request merged**